src/lsptoolshost/nestedCodeAction.ts

```diff
@@ -0,0 +1,131 @@
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { URIConverter, createConverter } from 'vscode-languageclient/lib/common/protocolConverter';
+import { UriConverter } from './uriConverter';
+import { getFixAllResponse } from './fixAllCodeAction';
+
+export function registerNestedCodeActionCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.OutputChannel
+) {
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'roslyn.client.nestedCodeAction',
+            async (request): Promise<void> => registerNestedResolveCodeAction(languageServer, request, outputChannel)
+        )
+    );
+}
+
+async function registerNestedResolveCodeAction(
+    languageServer: RoslynLanguageServer,
+    codeActionData: any,
+    outputChannel: vscode.OutputChannel
+): Promise<void> {
+    if (codeActionData) {
+        const data = <LSPAny>codeActionData;
+        const action = data.NestedCodeAction;
+
+        if (action?.nestedActions?.length > 0) {
+            const codeActionTitles: string[] = getCodeActionTitles(action.nestedActions);
+            const selectedValue = await vscode.window.showQuickPick(codeActionTitles, {
+                placeHolder: vscode.l10n.t(action.title),
+                ignoreFocusOut: true,
+            });
+            if (selectedValue) {
+                const selectedAction = retrieveSelectedAction(selectedValue, action.nestedActions);
+
+                if (!selectedAction) {
+                    return;
+                }
+
+                if (selectedAction.data.FixAllFlavors) {
+                    await getFixAllResponse(selectedAction.data, languageServer, outputChannel);
+                    return;
+                }
+
+                await vscode.window.withProgress(
+                    {
+                        location: vscode.ProgressLocation.Notification,
+                        title: vscode.l10n.t('Nested Code Action'),
+                        cancellable: true,
+                    },
+                    async (_, token) => {
+                        const nestedCodeActionResolve: CodeAction = {
+                            title: selectedAction.title,
+                            data: selectedAction.data,
+                        };
+
+                        const response = await languageServer.sendRequest(
+                            CodeActionResolveRequest.type,
+                            nestedCodeActionResolve,
+                            token
+                        );
+
+                        if (!response.edit) {
+                            outputChannel.show();
+                            outputChannel.appendLine(`Failed to make an edit for completion.`);
+                            return;
+                        }
+
+                        const uriConverter: URIConverter = (value: string): vscode.Uri =>
```

Comment:
Is it possible to use an instance of this instead?
https://github.com/dotnet/vscode-csharp/blob/6e35fe33f77060f651e81c27450479fc847a0156/src/lsptoolshost/uriConverter.ts#L9

Reply:
for what exactly?